homeassistant/helpers/entity_component.py

```diff
@@ -128,7 +128,7 @@ def setup(self, config: ConfigType) -> None:
             tasks.append(self.async_setup_platform(p_type, p_config))
 
         if tasks:
-            await asyncio.wait(tasks)
+            await asyncio.gather(*tasks)
```

Comment:
We shouldn't want this, because we don't want setup of 1 platform cancel the others. The same with the change for reset. We still want to log them.

Reply:
In all proposed changes here, it's ok to just log the errors and continue.

Reply:
that makes sense, but brings up a few questions:
1. If we don't want this specific exception to be caught, perhaps instead of throwing the exception, just logging? Is there anywhere that we do catch it and do something with it?
2. If my change passed all the unit tests, but there is functionality that it breaks (we don't want setup of 1 platform cancel the others), perhaps we should add a unit tests that tests this scenario. I don't know enough about the startup process of hass to create this, but sounds like a simple test to prevent others from making my mistake :)
3. In any case, even if we cancel this exception, probably better to change the wait(tasks) to gather(*tasks) so the exceptions don't disappear and catch them in the calling method. I tried looking and it seems like _async_set_up_integrations in bootstrap.py may be the right place, but not sure

Reply:
1. We should log it in this method where we change from wait to gather. Gather can return exceptions.
2. Yeah that's bad. I'll add one.
3. Yeah

Reply:
ok. just added the changes to _async_set_up_integrations 
I assume CORE_INTEGRATIONS don't need to be checked as if they fail, we fail anyway, right?